cmd/register.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/phomer/scheduler/accounts"
+)
+
+func main() {
+	username := accounts.Username()
+	hostname := accounts.Hostname()
+
+	fmt.Println("Registering ", username, " on ", hostname)
+
+	// Call the JWT stuff to generate the auth info
+	config := accounts.NewClientConfig(username, hostname)
+
+	auth := accounts.NewAuthentication()
+	account := accounts.NewAccount(username, hostname)
+
+	// Lock the Accounts database and update it
+	auth.Lock()
+	auth.Reload()
+	auth.UpdateAccount(account)
+	auth.Update()
+
+	// Release the lock
+	auth.Unlock()
```

Comment:
This seems like a very verbose API to update an account. Any reason why those steps are not inlined in `UpdateAccount` itself?

Reply:
I had buried them originally, but then I needed transactional integrity for Reload, UpdateAccount and Update, so I pushed the locks up to external. Later I realized that since there are really only 2 locations in the code that play here, the decomposition is wrong and it should all be internal (as I normally prefer).